src/Data/Hashable/Class.hs

```diff
@@ -763,7 +769,15 @@ hashThreadId (ThreadId t) = hash (fromIntegral (getThreadId t) :: Int)
 
 -- this cannot be capi, as GHC panics.
 foreign import ccall unsafe "rts_getThreadId" getThreadId
```

Comment:
is `rts_getThreadId` exported from some header? Could you change that to `capi` (with `CApiFFI`).

Reply:
Ah, crap. It cannot be `capi` as GHC panics, but does even GHC-9.4 panic still? It does :( https://gitlab.haskell.org/ghc/ghc/-/issues/22705

Reply:
If I change it to `capi`, I get
```
ghc: panic! (the 'impossible' happened)
  (GHC version 9.2.4:
	toCType
  ThreadId#
  Call stack:
      CallStack (from HasCallStack):
        callStackDoc, called at compiler/GHC/Utils/Panic.hs:181:37 in ghc:GHC.Utils.Panic
        pprPanic, called at compiler/GHC/HsToCore/Foreign/Decl.hs:763:17 in ghc:GHC.HsToCore.Foreign.Decl
```
which seems to be consistent with the comment

https://github.com/haskell-unordered-containers/hashable/blob/18138a173a7659e390ba4c026f9ceb1cd850b80e/src/Data/Hashable/Class.hs#L764

you added in #213.